kiwi/builder/disk.py

```diff
@@ -823,17 +834,21 @@ def _build_and_map_disk_partitions(
         else:
             rootfs_mbsize = 'all_free'
 
-        if self.volume_manager_name and self.volume_manager_name == 'lvm':
-            log.info('--> creating LVM root partition')
-            disk.create_root_lvm_partition(rootfs_mbsize)
-
-        elif self.mdraid:
-            log.info('--> creating mdraid root partition')
-            disk.create_root_raid_partition(rootfs_mbsize)
-
+        if self.root_filesystem_is_overlay and \
+           self.root_filesystem_has_write_partition is False:
+            log.warning(
+                '--> overlayroot explicitly requested no write partition'
+            )
         else:
-            log.info('--> creating root partition')
-            disk.create_root_partition(rootfs_mbsize)
+            if self.volume_manager_name and self.volume_manager_name == 'lvm':
```

Comment:
Why are you embedding this part as an `else`? Isn't it validated before, is it possible having `mraid` with overlay and write partition? This change of having this part under an `else` for the new use case feels odd or unneeded.

Reply:
Hmm, I'm not sure what you mean by that and maybe it is a bit hard to read how diff structured the change. If you take the code in question as a block it looks like this

```python
if self.root_filesystem_is_overlay and \
   self.root_filesystem_has_write_partition is False:
    log.warning(
                '--> overlayroot explicitly requested no write partition'
            )
   else:
       if self.volume_manager_name and self.volume_manager_name == 'lvm':
           log.info('--> creating LVM root partition')
           disk.create_root_lvm_partition(rootfs_mbsize)
       elif self.mdraid:
           log.info('--> creating mdraid root partition')
           disk.create_root_raid_partition(rootfs_mbsize)
       else:
           log.info('--> creating root partition')
           disk.create_root_partition(rootfs_mbsize)
```

from that perspective I think it makes sense. In overlayroot and without write partition we don't want an additional partition. Also note in overlayroot mode the ro partition is named rootro and the read write partition is named ```root```
In fact in an overlay setup the write partition is treated as the defacto root partition. The condition for creating this
partition or not is formulated in code like above

And back to your question, yes we also support the root overlay on top of a raid. It imho doesn't make sense but it is possible.

Does this makes more sense now ?

Reply:
Yes it does,  I got confused by the early check on volume_manager_name against root_filesystem_is_overlay.